tests/table/test_init.py

```diff
@@ -676,6 +676,15 @@ def test_update_metadata_set_snapshot_ref(table_v2: Table) -> None:
     )
 
 
+@pytest.mark.integration
+def test_cherrypick_snapshot(table_v2: Table):
+    cherrypick_operation = table_v2.transaction().cherrypick_snapshot(snapshot_id=3051729675574597004)
+    cherrypick_operation.cherrypick()
+    cherrypick_operation.commit()
```

Comment:
Not sure how to run/write this test. I get this error when I try to read the manifest files.

```../../pyiceberg/io/pyarrow.py:303: in create
    output_file = self._filesystem.open_output_stream(self._path, buffer_size=self._buffer_size)
pyarrow/_fs.pyx:881: in pyarrow._fs.FileSystem.open_output_stream
    ???
pyarrow/error.pxi:154: in pyarrow.lib.pyarrow_internal_check_status
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OSError: When initiating multiple part upload for key 'test/location/metadata/snap-6884214265407574614-0-f2de21d9-9512-4f23-b567-c5664d634609.avro' in bucket 'bucket': AWS Error ACCESS_DENIED during CreateMultipartUpload operation: Access Denied
```

I'm assuming that's because we're mocking the table metadata using conftest.py, and there are no manifest files or data files.

Reply:
resolved. Ran test as integration test locally after pulling and merging required changes that exist in other PRs.